MIPs/mip-x-request.md

```diff
@@ -0,0 +1,170 @@
+
+Generally, the implementation of a Multinetwork API will serve as a significant enabler of permissionless innovation in the community. We anticipate that it will drive  significant growth in existing and entirely new categories of apps.
+
+# Proposal
+
+## Language
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" written in uppercase in this document are to be interpreted as described in RFC 2119.
+
+## Definitions
+
+See the definitions outlined in [MIP-X](./mip-x-authorize.md)
+
+**routing parameter**: A parameter that is exclusively used for routing requests to a subsystem in MetaMask. The term `scope` may be used interchangably with "routing parameter" for the purpose of this MIP.
+
+## Proposal Specification
+
+### Multinetwork Wallet Request
+
+According to the CAIP-27 standard, a JSON-RPC request scoped to a particular chain should be made by wrapping JSON-RPC requests inside a `wallet_invokeMethod` JSON-RPC method. This `wallet_invokeMethod` method should contain an additional `scope` parameter that indicates where the request will be routed and a `request` parameter for the scope-specific JSON-RPC request.
+
+#### Example of a CAIP-27 compliant request
+
+##### Single Request
+Here is an example of how a method to query an Ethereum Mainnet balance would be wrapped in a `wallet_request`:
+
+```javascript
+provider.request({
+    "jsonrpc": "2.0",
+    "method": "wallet_invokeMethod",
+    "params":[{
+        "version": "0.0.1",
+        "scope": "eip155:1",
+        "request":{
+            "method": "eth_getBalance",
+            "params": ["0xabc...","latest"]
+        }
+    }]
+    "id": 1
+});
+```
+
+In this example, the `scope` field is populated with the [CAIP-2](https://chainagnostic.org/CAIPs/caip-2) identifier for Ethereum Mainnet, and the `request` field contains the typical JSON-RPC request for getting an Ethereum balance.
+
+##### Batch Request for Multiple Chains
+Here's an example of how JSON-RPC batch requests to get balances from both Ethereum Mainnet and Binance Smart Chain would be called by wrapping the requests in a `provider_request`:
+
+```javascript
+provider.request([
+  {
+    "jsonrpc": "2.0",
+    "method": "wallet_invokeMethod",
+    "params": {
+        "scope": "eip155:1",
+        "request": {
+            "method": "eth_getBalance",
+            "params": ["0xabc...", "latest"]
+      }
+    }
+    "id": 1
+  },
+  {
+    "jsonrpc": "2.0",
+    "method": "wallet_invokeMethod",
+    "params": {
+        "scope": "eip155:59144",
+        "request": {
+            "method": "eth_getBalance",
+            "params": ["0xabc...", "latest"]
+      }
+    }
+    "id": 2
+  }
+]);
```

Comment:
```suggestion
provider.request([
  {
    "id": 1,
    "jsonrpc": "2.0",
    "method": "wallet_invokeMethod",
    "params": {
        "scope": "eip155:1",
        "request": {
            "method": "eth_getBalance",
            "params": ["0xabc...", "latest"]
      }
    }
  },
  {
    "id": 2,
    "jsonrpc": "2.0",
    "method": "wallet_invokeMethod",
    "params": {
        "scope": "eip155:59144",
        "request": {
            "method": "eth_getBalance",
            "params": ["0xabc...", "latest"]
      }
    }
  }
]);
```

nit

Reply:
(moves the `id`s to the top)

Reply:
The [JSON-RPC 2.0 spec examples](https://www.jsonrpc.org/specification#examples) shows the `id` after `method` & `params`. Is there a reason to order them differently?